### PR TITLE
Delete conversation

### PR DESCRIPTION
**Pull request by KirstinBoni**
Added method to delete a conversation and unit tests.

### CODE REVIEW

File: test/codeu/chat/server/DeleteConversationTest.java [hunk 1]
```diff
@@ -84,4 +85,36 @@ public void testdeleteMultipleConversations() {
...
+  }
+  
+  @Test
+  public void testdeleteConversationDoesNotExistNonEmpty() {
+	  final User user = controller.newUser("user");
+	  final Conversation conversation = controller.newConversation("conversation", user.id);
+	  
+	  controller.deleteConversation("convo");
+	  
+	  assertFalse(model.conversationByText().first("conversation") == null);
```
Comment (bluedragon2) [inline test/codeu/chat/server/DeleteConversationTest.java:103]: Maybe assertTrue(... != null) ? That reads a bit more naturally. What do you think? Or better yet assertNotNull()?

File: test/codeu/chat/server/DeleteConversationTest.java [hunk 2]
```diff
@@ -84,4 +85,36 @@ public void testdeleteMultipleConversations() {
 	  assertTrue(model.conversationById().first(testConversation1.id) == null&& model.conversationById().first(testConversation2.id)==null);
```
Comment (bluedragon2) [inline test/codeu/chat/server/DeleteConversationTest.java]: Please break this line up. It is too long:
assertTrue(model.conversationById().first(testConversation1.id) == null 
  && model.conversationById().first(testConversation2.id)==null);

File: test/codeu/chat/server/DeleteConversationTest.java [hunk 3]
```diff
@@ -84,4 +85,36 @@ public void testdeleteMultipleConversations() {
...
+	  
+	  assertNull(model.conversationByText().first("conversation"));
+  }
+  
+  @Test
+  public void testdeleteConversationDoesNotExistNonEmpty() {
+	  final User user = controller.newUser("user");
+	  final Conversation conversation = controller.newConversation("conversation", user.id);
+	  
+	  controller.deleteConversation("convo");
```
Comment (bluedragon2) [inline test/codeu/chat/server/DeleteConversationTest.java:101]: What happens when you try to delete a non-existing conversation?

File: test/codeu/chat/server/DeleteConversationTest.java [hunk 4]
```diff
@@ -84,4 +85,36 @@ public void testdeleteMultipleConversations() {
...
+	  final User user = controller.newUser("user");
+	  final Conversation conversation = controller.newConversation("conversation", user.id);
+
+	  final User user2 = controller.newUser("user2");
+	  final Conversation conversation2 = controller.newConversation("conversation2", user.id);
+	  
+	  controller.deleteConversation("conversation2");
+	  
+	  assertNull(model.conversationByText().first("conversation2"));
+	  assertFalse(model.conversationByText().first("conversation") == null);
```
Comment (bluedragon2) [inline test/codeu/chat/server/DeleteConversationTest.java:117]: assertNotNull ?

File: test/codeu/chat/server/DeleteConversationTest.java (hunk 3 above)
Comment (KirstinBoni) [reply test/codeu/chat/server/DeleteConversationTest.java:101]: An error message is output alerting the client that me does not exist, this was handled from the client side so it was manually tested.

**Pull request merged**